packages/bsky/src/api/app/bsky/graph/getSuggestedFollowsByActor.ts

```diff
@@ -0,0 +1,122 @@
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+
+const MAX_RESULTS_LENGTH = 10
+
+export default function (server: Server, ctx: AppContext) {
+  server.app.bsky.graph.getSuggestedFollowsByActor({
+    auth: ctx.authVerifier,
+    handler: async ({ auth, params }) => {
+      const { actor } = params
+      const viewer = auth.credentials.did
+
+      const db = ctx.db.getReplica()
+      const actorService = ctx.services.actor(db)
+      const actorDid = await actorService.getActorDid(actor)
+
+      if (!actorDid) {
+        throw new InvalidRequestError('Actor not found')
+      }
+
+      const likes = await db.db
```

Comment:
i think it'd be better to do this aggregation in the db query itself to prevent 1000 rows from being sent back

ie
```ts
db.selectFrom(subquery).groupBy('subject').select('subject', 'count(*').orderBy('count', 'desc').limit(20)`
```

or similar, i'm sure my syntax is off

where the `subquery` is the same query you have here

Reply:
sick, love it, thanks

Reply:
So I tried this, but realized that we can't group directly by `subject` since it's a URI. We need to group by the `did` portion of the URI to find most commonly liked accounts.

We could maybe use a substring helper to do this in the db query itself, looks like that differes by postgres/sqlite 🤔

Reply:
Just posting that the query that I think should work is basically:

```ts
const mostLikedAccounts: { subject }[] = await db.db
  .selectFrom('like')
  .select('subject')
  .select(qb => qb.fn.count('subject').as('count')) // problem is subject is a URI
  .where('creator', '=', actorDid)
  .groupBy('subject')  // problem is subject is a URI
  orderBy('count', 'desc')
```

Reply:
(no need to worry about sqlite in the appview btw)

a small note on your query - i think it's still worth doing this in a subquery because then we're cutting off the total aggregation (only looking at 1000 rows max not the entire likeset)

but yeah that's a good point... lemme think

Reply:
off the cuff, a substring helper sounds like a pain - but at the same time, it should be pretty lightweight if it's wrapped in a subquery 🤔 

basically:
- grab 1000 most recent likes
- substring each of those uris to the subjectDid
- then run aggregation over that filtered set

Reply:
but this also all might just be too galaxy brained for the problem at hand - really 1000 rows (of just 2 dids)  isn't _that_ big

Reply:
okay final thoughts:

i think we can make this work fine with scylla (or at least something similar to it work fine)

if it's easy enough to write the substring in postgres, I think I would do that using the subquery (limited to 1000 still)

if not, then let's do it the old way and we _may_ have to drop the query size depending on performance. Is there a certain level at which you feel it loses signal? Like 500 likes seems fine? 200?